src/halmos/sevm.py

```diff
@@ -877,6 +921,26 @@ def call_unknown() -> None:
                 # vm.warp(uint256)
                 elif eq(arg.sort(), BitVecSort((4+32)*8)) and simplify(Extract(287, 256, arg)) == hevm_cheat_code.warp_sig:
                     ex.block.timestamp = simplify(Extract(255, 0, arg))
+                # vm.etch(address,bytes)
+                elif extract_funsig(arg) == hevm_cheat_code.etch_sig:
+                    # ex.pgm contains 256-bit keys
+                    who = simplify(ZeroExt(96, extract_bytes(arg, 4 + 12, 20)))
+
+                    # code must be concrete
+                    try:
+                        code_offset = extract_bytes(arg, 4 + 32, 32).as_long()
+                        code_length = extract_bytes(arg, 4 + code_offset, 32).as_long()
+                        code_int = extract_bytes(arg, 4 + code_offset + 32, code_length).as_long()
+                        code_bytes = code_int.to_bytes(code_length, 'big')
+
+                        pgm, code = decode_bytes(code_bytes)
+                        ex.pgm[who] = pgm
+                        ex.code[who] = code
```

Comment:
in theory, this could be unsound when `who` is symbolic, because it may alias with other existing address keys.  note that pgm and code are plain python dictionaries, not smt arrays, so no alias reasoning is done here.

the same issue exists even if `who` is concrete, as the `this` address is currently symbolic and may alias with it. perhaps, we should make the `this` address concrete, so that we can avoid alias reasoning at least for concrete `who`.

Reply:
disabled in f76f1841d